helpers.go

```diff
@@ -245,42 +245,57 @@ func getImageID(name string) (string, error) {
 
 // commandFunc represents a function that accepts an image ID and the CLI
 // context. This is used in the commandInContainer function.
-type commandFunc func(string, *cli.Context)
+type commandFunc func(string, *cli.Context) error
 
 // commandInContainer extracts the containerID from the given ctx. The function
 // first checks whether the container exists. If the container is running the
 // given commandFunc is executed on the image the container is based on. If the
 // --force flag is set the container is committed to a new image instead. In
 // case the container is not running it is committed to an image first.
 // Afterward commandFunc is executed on the new image.
-func commandInContainer(f commandFunc, ctx *cli.Context) {
+func commandInContainer(f commandFunc, ctx *cli.Context) (string, error) {
 	containerID := ctx.Args().First()
+	var image string
+	var err error
 	// check if the container exists
 	if !checkContainerExists(containerID) {
 		logAndFatalf("Container %s does not exist.\n", containerID)
 	}
 	// check whether the container is running
-	if container, err := checkContainerRunning(containerID); err != nil {
+	if container, runErr := checkContainerRunning(containerID); runErr != nil {
 		logAndPrintf("Checking stopped container %s ...\n", containerID)
 		// execute commitAndExecute on the stopped container
-		err = commitAndExecute(f, ctx, containerID)
-		if err != nil {
-			logAndFatalf("%v.\n", err)
-		}
+		image, err = commitAndExecute(f, ctx, containerID)
 	} else {
 		// if the force flag is set commitAndExecute is executed on the running container
 		if ctx.GlobalBool("force") {
 			logAndPrintf("WARNING: Force flag used. Manually installed packages will be analyzed as well.\n")
-			err = commitAndExecute(f, ctx, containerID)
-			if err != nil {
-				logAndFatalf("%v.\n", err)
-			}
+			image, err = commitAndExecute(f, ctx, containerID)
 		} else {
 			// directly call commandFunc on the image the container is based on
 			logAndPrintf("WARNING: Only the source image from this container will be inspected. Manually installed packages won't be taken into account.\n")
-			f(container.Image, ctx)
+			err = f(container.Image, ctx)
+			image = container.Image
+		}
+	}
+	return image, err
+}
+
+// check whether the given error has a zypper exit code != 0 and decide if it is severe
+func checkError(image, cmd string, err error) {
+	if err == nil {
+		return
+	}
+	switch err.(type) {
+	case dockerError:
+		de := int(err.(dockerError).exitCode)
+		if !isZypperExitCodeSevere(de) {
+			exitWithCode(de)
+			return
```

Comment:
Right, some of the tests fail, if i remove the return here, but I do not understand why. This is why I left it in.

Reply:
I get it. The tests fails without the return, because the function is overwritten for the tests, but for normal execution it is not reachable. 
For now I will remove the return and update the test that fails to return exit code 1 instead of 100.